### PR TITLE
LLM(Large Language Model) Python sample

### PR DESCRIPTION
**Pull request by wgzintel**
Details:
The python example that learn how to estimate performance for LLM. It is based on pipelines provided by Optimum-Intel and allows to estimate performance for Pytorch and OpenVINO models using same pipelines.

Ticket ID: CVS-120025

### CODE REVIEW

**Comment by jiwaszki** [comment]
Isn't it the same as https://github.com/openvinotoolkit/openvino/pull/20111 ? Should one be closed in favor of another?

**Comment by peterchen-intel** [comment]
> Isn't it the same as #20111 ? Should one be closed in favor of another?

Yes. They are similar, but this one have updating content. One of them will be closed.

**Comment by peterchen-intel** [comment]
@p-wysocki Regarding https://github.com/openvinotoolkit/openvino/pull/20111#pullrequestreview-1664016260 
1. Semaev, Yuri has helped to get the approval.
2. Done - merged requirements-conversion.txt and requirements.txt to top folder requirements.txt
3. Done - merged requirements-conversion.txt and requirements.txt to top folder requirements.txt
4. Checked, there is no duplicate constraints by comparing to src/bindings/python/constraints.txt, tests/constraints.txt, and tools/constraints.txt

File: samples/python/llm/requirements.txt
Comment (peterchen-intel) [inline samples/python/llm/requirements.txt:1]:
```suggestion
--extra-index-url https://download.pytorch.org/whl/cpu
numpy
```

File: samples/python/llm/requirements.txt
Comment (peterchen-intel) [inline samples/python/llm/requirements.txt:2]: @jiwaszki Updated to use openvino now.

File: samples/python/llm/requirements.txt
Comment (peterchen-intel) [inline samples/python/llm/requirements.txt:9]: @jiwaszki Since the LLM related packages are being developed swiftly and being changed fast, we have to use a specific commit ID or even the HEAD of master to support as many models as possible, is it OK to do in this way?

File: samples/python/llm/utils/ov_model_classes.py
Comment (peterchen-intel) [inline samples/python/llm/utils/ov_model_classes.py:109]: @eaidova Can you answer the question in https://github.com/openvinotoolkit/openvino/pull/20111/files#r1350183118 from @jiwaszki ?
"Is there any explicit reason to use asynchronous API, any perf benefits?"

**Review by peterchen-intel — APPROVED** [review]

**Review by WeldonWangwang — APPROVED** [review]

File: samples/python/llm/utils/hook_transformers.py
Comment (xipingyan) [inline samples/python/llm/utils/hook_transformers.py:190]: Update all time statistic functions to be consistent with `time.perf_counter()`.

File: samples/python/llm/requirements.txt
Comment (jiwaszki) [reply samples/python/llm/requirements.txt:3]: Great, thanks!

File: samples/python/llm/requirements.txt
Comment (jiwaszki) [reply samples/python/llm/requirements.txt:10]: It is absolutely understandable now, resolving it.

**Review by jiwaszki — APPROVED** [review]
I give it a green light but it will be great to have the understanding of "async over sync" benefits.

File: docs/articles_en/learn_openvino/openvino_samples/python_sample_large_language_model.md
Comment (kblaszczak-intel) [inline docs/articles_en/learn_openvino/openvino_samples/python_sample_large_language_model.md:21]:
```suggestion
(1) Start a Python virtual environment
```

File: docs/articles_en/learn_openvino/openvino_samples/python_sample_large_language_model.md
Comment (kblaszczak-intel) [inline docs/articles_en/learn_openvino/openvino_samples/python_sample_large_language_model.md:29]:
```suggestion
(2) Convert the model to OPENVINO IR
```

File: docs/articles_en/learn_openvino/openvino_samples/python_sample_large_language_model.md
Comment (kblaszczak-intel) [inline docs/articles_en/learn_openvino/openvino_samples/python_sample_large_language_model.md:46]:
```suggestion
Parameters:
```

**Review by kblaszczak-intel — APPROVED** [review]
The article itself does require some improvements. But the formatting and typos can be fixed with a followup PR. We'll take care of it after this one is merged.

**Review by xipingyan — APPROVED** [review]

File: docs/articles_en/learn_openvino/openvino_samples/python_sample_large_language_model.md
Comment (WeldonWangwang) [inline docs/articles_en/learn_openvino/openvino_samples/python_sample_large_language_model.md:38]: requirements_conversion.txt has been renamed.

**Comment by wenjiew** [comment]
@peterchen-intel @wgzintel Any additional commits required for this PR? Thanks!

**Comment by peterchen-intel** [comment]
@wenjiew This PR is ready for merge, we can create additional PR if there is any new requirements.

**Comment by peterchen-intel** [comment]
As communicated, the sample is being moved to https://github.com/openvinotoolkit/openvino.genai/pull/7

**Pull request closed without merging**
